lib/Common/CommonPal.h

```diff
@@ -611,19 +611,20 @@ HRESULT ULongMult(ULONG ulMultiplicand, ULONG ulMultiplier, ULONG* pulResult);
 template <class TryFunc, class FinallyFunc>
 void TryFinally(const TryFunc& tryFunc, const FinallyFunc& finallyFunc)
 {
-    bool hasException = true;
-    try
+    class FinallyObject
```

Comment:
>class FinallyObject [](start = 4, length = 19)

Still wondering why rethrow in catch can cause stackoverflow and how could destructor avoid that?